Docs: unsupported floating grids in asymmetric calculations (#1367)
PowerGridModel/power-grid-model: docs/user_manual/calculations.md

- Node voltage magnitude and angle
- Current flowing through branches and fault.

```{note}

> Short-circuit calculations are currently implemented in the phase (abc) domain and therefore require a grounded network, similar to asymmetric power flow calculations.

Reviewer: this is true for all except for symmetric faults (ABC). For those, we short-cut and do symmetric calculations, but i do not know whether our current official support and data validator follow that assumption as well. Needs to be double-checked

Author: Both of the support and data validator are also tested before revising the documentation.

In power-grid-model, a network is considered grounded if at least one of the following is present:
- A transformer with grounded neutral (e.g. `winding_from`/`winding_to` of a transformer is set to `wye_n`, `zigzag_n`).
- A shunt connected to ground.
- Any other component that provides a conductive path to ground.

> A future implementation using the sequence (0-1-2) domain is expected to remove this limitation.

Reviewer 2: This is only for short circuit calculations, right?
```suggestion
A future implementation using the sequence (0-1-2) domain is expected to remove this limitation for short circuit calculations.
```

> A future implementation using the sequence (0-1-2) domain is expected to remove this limitation for short circuit calculations.

Reviewer: idem, optional

```suggestion
A potential future implementation using the sequence (0-1-2) domain is expected to remove this limitation for short circuit calculations.
```

- Node voltage magnitude and angle
- Current flowing through branches and fault.

```{note}
Short-circuit calculations are currently implemented in the phase (abc) domain and therefore require a grounded network, similar to asymmetric power flow calculations.  

> A future implementation using the sequence (0-1-2) domain is expected to remove this limitation.

Reviewer 2: Should we add in this note a "suggestion" like: grounding can happen either through a Yn/Dn transformer or by a shunt. Or is this common EE knowledge? Edit: I see this is already mentioned below. Since this block comes first, perhaps either quote with hyperlink here or just mention here and quote there?

On the other hand, should we also make a note of this in the short circuit algorithm details (i.e. https://power-grid-model.readthedocs.io/en/stable/algorithms/sc-algorithms.html), perhaps with more detail (or not)?

> A future implementation using the sequence (0-1-2) domain is expected to remove this limitation.

Reviewer 3: Although approaching via sequence domain can solve this problem, there would not be a future implementation. This is a design choice of PGM. Hence best to remove it.

Author: Removed

  For asymmetric calculations voltages are given as line-to-neutral and output contains arrays with values per phase
  for all output variables.

```{note}

> In power-grid model, asymmetric calculations with certain configurations require the network to have a reference to ground. For details on how floating grids are treated in power-grid-model, please refer to[Floating grid handling](calulations.md#floating-grid-handling).

Reviewer 2: Small typo
```suggestion
In power-grid model, asymmetric calculations with certain configurations require the network to have a reference to ground. For details on how floating grids are treated in power-grid-model, please refer to [Floating grid handling](calulations.md#floating-grid-handling).
```

```

```{note}
For short-circuit calculations, a three-phase `fault_type` is calculated with a symmetric calculation, while any other
`fault_type` (e.g. single- or two-phase faults) automatically triggers the asymmetric calculation.
Outputs for short circuit calculations always give asymmetric output, independent of the fault type present.
```

#### Floating grid handling
In power-grid-model, two different concepts should be distinguished:

- Physical grounding of the network
  Whether the electrical system has an explicit path to ground (e.g. via transformer winding connection, shunts, or grounding elements).
- Numerical solvability in PGM
  Whether the formulation provides enough reference to compute a unique solution, even if the physical system is not explicitly grounded.

A floating grid issue only arises in specific configurations where:

- A transformer is present, and
- All involved windings are ungrounded (no star-point grounding, no delta grounding reference, etc.), and
- No other grounding path (shunt, source grounding, etc.) exists in the network.

> In this case, the system may lack a reference for certain sequence components (typically zero-sequence), leading to an ill-posed or singular system.

Reviewer 3: ```suggestion
In this case, the system may lack a reference for calculating line to ground voltages, leading to an ill-posed or singular system.
```

> In current power-grid-model, a shunt with small admittance is added only in transformer-related configurations where a grounding reference is missing. This shunt is connected to one side of the transformer. This is intended to ensure numerical stability in cases where the transformer topology introduces an ungrounded subsystem.

Reviewer 2: Minor for clarity
```suggestion
Currently in power-grid-model, a shunt with small admittance is added only in transformer-related configurations where a grounding reference is missing. This shunt is connected to one side of the transformer. This is intended to ensure numerical stability in cases where the transformer topology introduces an ungrounded subsystem.
```

> In current power-grid-model, a shunt with small admittance is added only in transformer-related configurations where a grounding reference is missing. This shunt is connected to one side of the transformer. This is intended to ensure numerical stability in cases where the transformer topology introduces an ungrounded subsystem.

Reviewer 3: ```suggestion
In current power-grid-model, a shunt with small admittance is added only in transformer-related configurations where a grounding reference is missing. This shunt is connected to one side of the transformer. This is intended to ensure numerical solvability in cases where the transformer topology introduces an ungrounded subsystem.
```

  For asymmetric calculations voltages are given as line-to-neutral and output contains arrays with values per phase
  for all output variables.

```{note}
Asymmetric calculations require the network to have a reference to ground. For the definition of a grounded grid, please refer to [Short circuit calculations](calulations.md#short-circuit-calculations).
If no such connection exists, the calculation cannot be solved and will fail with a sparse matrix error.

> Currently, power-grid-model addresses this scenario internally by adding a small admittance (implemented as a shunt) to one end of transformers to ensure the network has a reference to ground.

Reviewer 2: This reads as if the user creates by mistake a floating grid, we can handle it anyway because we internally would add a shunt with small admittance. Can you specify when this is done?

Reviewer: we only add a shunt with small admittance when a transformer is added, right? a grid with only a source and a load is still floating

  For asymmetric calculations voltages are given as line-to-neutral and output contains arrays with values per phase
  for all output variables.

```{note}

> In power-grid-model, a network is considered grounded if at least one of the following is present:

Reviewer 2: Maybe good to also add a note about this to the algorithm details? Not sure, but mentioning for consideration.

- A transformer with grounded neutral (e.g. `winding_from`/`winding_to` of a transformer is set to `wye_n`, `zigzag_n`)
- A shunt connected to ground
- Any other component that provides a conductive path to ground

Asymmetric calculations require the network to have a reference to ground.  
If no such connection exists, the calculation cannot be solved and will fail with a sparse matrix error.

> Our current data validator doesn't cover this check.

Reviewer 2: Do we have an issue for this?

Author: I'll rewrite this part based on test results.
